src/SecTester.Scan/ScanSettings.cs

```diff
@@ -0,0 +1,199 @@
+      }
+
+      _poolSize = value;
+    }
+  }
+
+  public TimeSpan? SlowEpTimeout
+  {
+    get { return _slowEpTimeout; }
+    set
+    {
+      if (value is null || value < MinSlowEpTimeout)
+      {
+        throw new ArgumentException("Invalid slow entry point timeout.");
+      }
+
+      _slowEpTimeout = value;
+    }
+  }
+
+  public TimeSpan? TargetTimeout
+  {
+    get { return _targetTimeout; }
+    set
+    {
+      if (value is null || (value < MinTargetTimeout || value > MaxTargetTimeout))
+      {
+        throw new ArgumentException("Invalid target connection timeout.");
+      }
+
+      _targetTimeout = value;
+    }
+  }
+
+  public bool? SkipStaticParams { get; set; }
+
+  public IEnumerable<TestType> Tests
+  {
+    get { return _tests; }
+    set
+    {
+      AssertEnum(value, TestTypeWhiteList,
+        "Unknown test type supplied.",
+        "Please provide at least one test.");
+
+      _tests = value.Distinct();
+    }
+  }
+
+  public IEnumerable<AttackParamLocation>? AttackParamLocations
+  {
+    get { return _attackParamLocations; }
+    set
+    {
+      AssertEnum(value, AttackParamLocationWhiteList,
+        "Unknown attack param location supplied.",
+        "Please provide at least one attack parameter location.");
+
+      _attackParamLocations = value!.Distinct();
+    }
+  }
+
+  public ScanSettings(TargetOptions targetOptions, IEnumerable<TestType> tests, string? repeaterId = default,
+    string? name = default)
+    : this(targetOptions, tests, repeaterId, name, DefaultSmart, DefaultPoolSize, DefaultTargetTimeout,
+      DefaultSlowEpTimeout, DefaultSkipStaticParams, DefaultAttackParamLocations)
+  {
+  }
+
+  internal ScanSettings(ScanSettingsOptions scanSettingsOptions)
+    : this(scanSettingsOptions.Target, scanSettingsOptions.Tests, scanSettingsOptions.RepeaterId,
+      scanSettingsOptions.Name, scanSettingsOptions.Smart, scanSettingsOptions.PoolSize,
+      scanSettingsOptions.TargetTimeout, scanSettingsOptions.SlowEpTimeout, scanSettingsOptions.SkipStaticParams,
+      scanSettingsOptions.AttackParamLocations)
+  {
+  }
+
+  private ScanSettings(TargetOptions targetOptions, IEnumerable<TestType> tests, string? repeaterId,
+    string? name,
+    bool? smart, int? poolSize, TimeSpan? targetTimeout,
+    TimeSpan? slowEpTimeout, bool? skipStaticParams,
+    IEnumerable<AttackParamLocation>? attackParamLocations)
+  {
+    Target = targetOptions;
+    Tests = tests;
+    Name = name ?? CreateDefaultName(targetOptions);
+    RepeaterId = repeaterId;
+    Smart = smart ?? DefaultSmart;
+    PoolSize = poolSize ?? DefaultPoolSize;
+    TargetTimeout = targetTimeout ?? DefaultTargetTimeout;
+    SlowEpTimeout = slowEpTimeout ?? DefaultSlowEpTimeout;
+    SkipStaticParams = skipStaticParams ?? DefaultSkipStaticParams;
+    AttackParamLocations = attackParamLocations ?? DefaultAttackParamLocations;
+  }
+
+  private static string CreateDefaultName(TargetOptions target)
```

Comment:
`TargetOptions` shouldn't be used to retrieve `Url` (URL must be normalized). It is already done in `Target`